src/components/cc-token-api-creation-form/cc-token-api-creation-form.smart.js

```diff
@@ -0,0 +1,168 @@
+        state.type = 'creating';
+      });
+      const componentState = /** @type {TokenApiCreationFormStateCreating} */ (component.state);
+
+      api
+        .createApiToken({ name, description, expirationDate, password, mfaCode })
+        .then((token) => {
+          /** @type {TokenApiCreationFormStateLoadedCopy} */
+          const newState = {
+            ...componentState,
+            type: 'loaded',
+            activeStep: 'copy',
+            token,
+          };
+
+          updateComponent('state', newState);
+        })
+        .catch(
+          /** @param {Error} error */
+          (error) => {
+            const errorCode =
+              'responseBody' in error && typeof error.responseBody === 'object' && 'code' in error.responseBody
+                ? error.responseBody.code
+                : null;
+            /** @type {TokenApiCreationFormStateLoadedValidation['credentialsError']} */
+            let credentialsError;
+
+            if (errorCode === 'invalid-credential') {
+              credentialsError = 'password';
+            }
+
+            if (errorCode === 'invalid-mfa-code') {
+              credentialsError = 'mfaCode';
+            }
+
+            /** @type {TokenApiCreationFormStateLoadedValidation} */
+            const newState = {
+              ...componentState,
+              type: 'loaded',
+              credentialsError,
+            };
+
+            updateComponent('state', newState);
+
+            if (credentialsError == null) {
+              notifyError(i18n('cc-token-api-creation-form.validation-step.error.generic'));
+            }
+          },
+        );
+    });
+  },
+});
+
+class Api {
+  /** @param {ApiConfig} apiConfig */
+  constructor(apiConfig) {
+    this._apiConfig = apiConfig;
+
+    /** @type {string|null} */
+    this._userEmail = null;
+  }
+
+  /**
+   * @param {object} options
+   * @param {string} options.name
+   * @param {string} options.description
+   * @param {string} options.expirationDate
+   * @param {string} options.password
+   * @param {string} options.mfaCode
+   */
+  _prepareCreateApiTokenRequest({ name, description, expirationDate, password, mfaCode }) {
+    return Promise.resolve({
+      method: 'post',
+      url: '/api-tokens',
+      headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
+      body: { email: this._userEmail, password, mfaCode, name, description, expirationDate },
+    });
+  }
+
+  /**
+   * @param {object} options
+   * @param {string} options.name
+   * @param {string} options.description
+   * @param {string} options.expirationDate
+   * @param {string} options.password
+   * @param {string} options.mfaCode
+   */
+  createApiToken({ name, description, expirationDate, password, mfaCode }) {
+    return this._prepareCreateApiTokenRequest({
+      password,
+      mfaCode,
+      name,
+      description,
+      expirationDate: dateFormatterIso.format(new Date(expirationDate)),
```

Comment:
the given `expirationDate` should already be an ISO date. Formatting should only be a UI implementation detail

Reply:
@pdesoyres-cc I chose to send a `Date` within the event so is it okay if I keep this within the smart? I feel like it's the smart's job to match the API expectation so it's ok?
(I could have gone for ISO strings within the event but I would have had to make the conversion within the UI component anyway so it's more or less the same? Not sure, feel free to tell me if you think we should go with ISO strings in the event :+1:)

Reply:
@pdesoyres-cc nevermind... I refactored to use `string` (ISO format) everywhere because it's a lot simpler since it's what the `cc-input-date` event payload contains :angel: